src/libraries/System.Runtime/tests/System/Reflection/NullabilityInfoContextTests.cs

```diff
@@ -1089,6 +1089,20 @@ public void TestNestedGenericInheritanceWithMultipleParameters()
             Assert.Equal(NullabilityState.NotNull, item3Info.ElementType!.ReadState);
             Assert.Equal(NullabilityState.NotNull, item3Info.ElementType.WriteState);
         }
+
+        [Fact]
+        [SkipOnMono("Nullability attributes trimmed on Mono")]
+        public void TestNullabilityInfoCreationOnPropertiesWithNestedGenericTypeArguments()
+        {
+            Type type = typeof(TypeWithPropertiesNestingItsGenericTypeArgument<int>);
+
+            Assert.NotNull(nullabilityContext.Create(type.GetProperty("Shallow1")!));
+            Assert.NotNull(nullabilityContext.Create(type.GetProperty("Deep1")!));
+            Assert.NotNull(nullabilityContext.Create(type.GetProperty("Deep2")!));
+            Assert.NotNull(nullabilityContext.Create(type.GetProperty("Deep3")!));
+            Assert.NotNull(nullabilityContext.Create(type.GetProperty("Deep4")!));
+            Assert.NotNull(nullabilityContext.Create(type.GetProperty("Deep5")!));
```

Comment:
Its better to assert the nullability result too, if the call produce correct result

Reply:
I added the checks. Unfortunately I could not build them on my local machine (I get weird errors on build that I did not use to get before)---but in my trimmed tests they went fine.